### PR TITLE
feat(ldap.js): support ldap.js createClient

### PR DESCRIPTION
**Pull request by itaykat**

### CODE REVIEW

File: test/unit_tests/events/ldapjs.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+require('dotenv').config();
+const ldap = require('ldapjs');
+// const { spawn } = require('child_process');
+const epsagon = require('../../../src/index');
+
+
+describe('ldap.js client events tests', () => {
+    // let server;
```
Comment (haimrait) [inline test/unit_tests/events/ldapjs.js]: remove comments

File: test/unit_tests/events/ldapjs.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    before((done) => {
+        // server = spawn('node', ['node_modules/ldap-server-mock/server.js',
```
Comment (haimrait) [inline test/unit_tests/events/ldapjs.js]: remove comments

File: test/unit_tests/events/ldapjs.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        epsagon.init({
+            token: 'my-secret-token',
+            appName: 'my-app-name',
+            metadataOnly: false,
+        });
+        done();
+    });
+
+    after((done) => {
+        // server.kill('SIGKILL');
```
Comment (haimrait) [inline test/unit_tests/events/ldapjs.js]: same remove comment

File: src/events/ldap.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                utils.createTimestampFromTime(startTime),
+                null,
+                'ldap',
+                0,
+                errorCode.ErrorCode.OK,
+            ]);
+            bindEvent.setResource(resource);
+            eventInterface.addToMetadata(bindEvent, {
+                'LDAP Client': {
+                    URL: url || '',
```
Comment (haimrait) [inline src/events/ldap.js]: if url = '' so why do you write URL || '' it seems useless

Comment (itaykat) [reply src/events/ldap.js]: It was a placeholder, I've implemented now the url from the client.

File: src/events/ldap.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+ * @param {Function} bindFunction The wrapped bind function from ldap.js module
+ * @returns {Function} The wrapped function
+ */
+function bindWrapper(bindFunction) {
+    return function internalBindWrapper(dn, password, controls, callback) {
+        try {
+            utils.debugLog(`LDAP.js bind() wrapper - dn: ${dn}`);
+
+            const resource = new serverlessEvent.Resource([
+                this.url,
```
Comment (galbash) [inline src/events/ldap.js]: consider using only the domain instead of the full URL as the resource name

Comment (galbash) [reply src/events/ldap.js]: (or if that's what the url already is that's fine)

File: src/events/ldap.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                    connectTimeout: this.connectTimeout || '',
+                    tlsOptions: this.tlsOptions || '',
+                    idleTimeout: this.idleTimeout || '',
+                    strictDN: this.strictDN || '',
+                    Controls: controls || '',
+                    DN: dn || '',
+                },
+            });
+            const responsePromise = new Promise((resolve) => {
+                callback = (err, res) => { // eslint-disable-line no-param-reassign
```
Comment (galbash) [inline src/events/ldap.js]: I would use a different parameter and not override the callback parameter

Comment (galbash) [reply src/events/ldap.js]: (initialize it to `callback` and override here)

File: src/events/ldap.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                'LDAP Client': {
+                    URL: this.url || '',
+                    socketPath: this.socketPath || '',
+                    timeout: this.timeout || '',
+                    connectTimeout: this.connectTimeout || '',
+                    tlsOptions: this.tlsOptions || '',
+                    idleTimeout: this.idleTimeout || '',
+                    strictDN: this.strictDN || '',
+                    Controls: controls || '',
+                    DN: dn || '',
```
Comment (galbash) [inline src/events/ldap.js]: let's talk about tags naming

File: src/events/ldap.js (hunk 5 above)
Comment (itaykat) [reply src/events/ldap.js]: it is only the domain

File: src/events/ldap.js (hunk 7 above)
Comment (itaykat) [reply src/events/ldap.js]: let me know when you want to talk, at your time.

**Review by galbash — APPROVED** [review]

**Review by galbash — APPROVED** [review]

**Pull request merged**